packages/yew/src/dom_bundle/btag/attributes.rs

```diff
@@ -83,21 +84,26 @@ impl Apply for InputFields {
 
 impl Attributes {
     #[cold]
-    fn apply_diff_index_maps<'a, A, B>(
+    fn apply_diff_index_maps<'a>(
         el: &Element,
         // this makes it possible to diff `&'a IndexMap<_, A>` and `IndexMap<_, &'a A>`.
-        mut new_iter: impl Iterator<Item = (&'static str, &'a str)>,
-        new: &IndexMap<&'static str, A>,
-        old: &IndexMap<&'static str, B>,
-    ) where
-        A: AsRef<str>,
-        B: AsRef<str>,
-    {
+        mut new_iter: impl Iterator<Item = (&'a str, &'a str)>,
+        new: &IndexMap<AttrValue, AttrValue>,
+        old: &IndexMap<AttrValue, AttrValue>,
+    ) {
         let mut old_iter = old.iter();
+        let new = new
+            .iter()
+            .map(|(k, v)| (k.as_ref(), v))
+            .collect::<IndexMap<_, _>>();
+        let old = old
+            .iter()
+            .map(|(k, v)| (k.as_ref(), v))
+            .collect::<IndexMap<_, _>>();
```

Comment:
If you `impl Borrow<str> for AttrValue`, then you don't need to recreate these maps (`new` and `old`). In my experiments, this accounts for the whole codesize change. The `IndexMap` API (like the `HashMap` API) allows to pass in types that "borrow as-if" they were the key type, here `str` instead of `AttrValue`. I'd also add a comment to, or even change, the `Hash` impl to forward to the `Borrow<str>`.